x-pack/plugin/sql/src/test/resources/org/elasticsearch/xpack/sql/planner/querytranslator_tests.txt

```diff
@@ -0,0 +1,826 @@
+{"geo_distance":{"point":[10.0,20.0],"distance":25.0,"distance_type":"arc","validation_method":"STRICT
+;
+
+StX
+SELECT ST_AsWKT(point) FROM test WHERE ST_X(point) = 10;
+InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.eq(InternalSqlScriptUtils.stX(InternalSqlScriptUtils.geoDocValue(doc,params.v0)),params.v1))
+"params":{"v0":"point","v1":10}
+;
+
+StY
+SELECT ST_AsWKT(point) FROM test WHERE ST_Y(point) = 10;
+InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.eq(InternalSqlScriptUtils.stY(InternalSqlScriptUtils.geoDocValue(doc,params.v0)),params.v1))
+"params":{"v0":"point","v1":10}
+;
+
+StGeometry
+SELECT ST_AsWKT(point) FROM test WHERE ST_GEOMETRYTYPE(point) = 'POINT';
+InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.eq(InternalSqlScriptUtils.stGeometryType(InternalSqlScriptUtils.geoDocValue(doc,params.v0)),params.v1))
+"params":{"v0":"point","v1":"POINT"}
+;
+
+
+// Top Hits
+///////////
+TopHitsAggregationWithOneArg1
+SELECT FIRST(keyword) FROM test;
+"top_hits":{"from":0,"size":1,"version":false,"seq_no_primary_term":false,
+"explain":false,"docvalue_fields":[{"field":"keyword"}],
+"sort":[{"keyword":{"order":"asc","missing":"_last","unmapped_type":"keyword"}}]}}}}}
+;
+
+TopHitsAggregationWithOneArg2
+SELECT MIN(keyword) FROM test;
+"top_hits":{"from":0,"size":1,"version":false,"seq_no_primary_term":false,
+"explain":false,"docvalue_fields":[{"field":"keyword"}],
+"sort":[{"keyword":{"order":"asc","missing":"_last","unmapped_type":"keyword"}}]}}}}}
+;
+
+TopHitsAggregationWithOneArg3
+SELECT LAST(date) FROM test;
+"top_hits":{"from":0,"size":1,"version":false,"seq_no_primary_term":false,
+"explain":false,"docvalue_fields":[{"field":"date","format":"strict_date_optional_time_nanos"}],
+"sort":[{"date":{"order":"desc","missing":"_last","unmapped_type":"date"}}]}}}}}
+;
+
+TopHitsAggregationWithOneArg4
+SELECT MAX(keyword) FROM test;
+"top_hits":{"from":0,"size":1,"version":false,"seq_no_primary_term":false,
+"explain":false,"docvalue_fields":[{"field":"keyword"}],
+"sort":[{"keyword":{"order":"desc","missing":"_last","unmapped_type":"keyword"}}]}}}}}
+;
+
+TopHitsAggregationWithTwoArgs1
+SELECT FIRST(keyword, int) FROM test;
+"top_hits":{"from":0,"size":1,"version":false,"seq_no_primary_term":false,
+"explain":false,"docvalue_fields":[{"field":"keyword"}],
+"sort":[{"int":{"order":"asc","missing":"_last","unmapped_type":"integer"}},
+{"keyword":{"order":"asc","missing":"_last","unmapped_type":"keyword"}}]}}}}}
+;
+
+TopHitsAggregationWithTwoArgs2
+SELECT LAST(date, int) FROM test;
+"top_hits":{"from":0,"size":1,"version":false,"seq_no_primary_term":false,
+"explain":false,"docvalue_fields":[{"field":"date","format":"strict_date_optional_time_nanos"}],
+"sort":[{"int":{"order":"desc","missing":"_last","unmapped_type":"integer"}},
+{"date":{"order":"desc","missing":"_last","unmapped_type":"date"}}]}}}}}
+;
+
+
+// Subqueries
+/////////////
+SubqueryBasicSelect
+SELECT int FROM
+    (SELECT int FROM test);
+;
```

Comment:
These queries are checked only for validity not for translation correct?
If so it makes sense to move them to a different file, to indicate a different type of check.

Reply:
Sure, yep, all the subquery related are just checked that are correctly analysed planned and optimised.

Reply:
For the EQL tests that currently reside in one file, I use only the test name as it was, i.e.:
<img width="533" alt="Screen Shot 2021-04-22 at 11 48 39" src="https://user-images.githubusercontent.com/5058131/115685287-61e85e00-a358-11eb-95a6-a17ece484a54.png">

For the SQL tests I use a `/` to separate the filename from the test name, i.e.:
<img width="638" alt="Screen Shot 2021-04-22 at 11 50 25" src="https://user-images.githubusercontent.com/5058131/115685500-90663900-a358-11eb-9915-bebec2de8ffb.png">

If you think that causes any trouble with running tests from inside IDE, let me know.

Reply:
Reproduction cmd line looks like this:
```
./gradlew ':x-pack:plugin:sql:test' --tests "org.elasticsearch.xpack.sql.planner.QueryTranslatorSpecTests" -Dtests.method="test {querytranslator_tests.txt/NoCountDoesNotTrackHits}" -Dtests.seed=BF93038714D8F6B -Dtests.locale=en-IE -Dtests.timezone=Africa/Sao_Tome -Druntime.java=16
```